### ISSUE TITLE
Notifications: can't follow a user with non-typical domain name

### ISSUE THREAD

**Issue by lancewillett** [Automattic/wp-calypso#39993]
This seems to happen when a site doesn’t _appear_ to be on WordPress.com. Such as a Simple site with a `.blog` custom domain. Or, when the user's account "Primary Blog" setting in Calypso is not a WordPress.com site, such as a Twitter profile.

#### Steps to reproduce
1. Starting at any logged-in URL
2. Open notifications
3. Look for blank "follow" links next to WordPress.com users who've liked or commented on your site content

#### What I expected

To see a follow link for any WordPress.com site in another users' profile.

<img width="673" alt="Screen Shot 2020-03-09 at 14 51 50" src="https://user-images.githubusercontent.com/66797/76260653-8301f400-6215-11ea-96ad-7cfc5dfe215a.png">

Jason Caldwell’s link goes to Twitter — so that's OK, you can't follow it in Reader (yet!). Ballio's site is on a .blog like me — and the Follow action link isn’t present for `https://ballio.blog/`.

Here's an example from @rralian where he sees my WordPress.com URL but not a follow button. `https://sensible.blog/` is a WordPress.com Simple site with a .blog custom domain.

<img width="555" alt="Screen Shot 2020-03-09 at 14 52 10" src="https://user-images.githubusercontent.com/66797/76260662-8eedb600-6215-11ea-9c5c-f4a1148dab5c.png">

#### What happened instead

There is an empty "link" area with nothing to follow by clicking.

#### Notes

I tracked it down a bit to `FollowLink` component — which is what controls that “Follow” icon and action under each user in the notifications list.

Source code: `calypso/apps/notifications/src/panel/ templates/block-user.jsx#137`
https://github.com/Automattic/wp-calypso/blob/master/apps/notifications/src/panel/templates/block-user.jsx#L137

Needs more debugging: CC @dmsnell 

What data is `this.props.block.meta.ids.site` pulling from? Is it the primary blog? Why would it be undefined for a WordPress.com Simple site with a custom .blog domain?

**Comment by dmsnell** [comment]
@lancewillett it looks like we limited these in r117068-wpcom to users who were members of blogs with these kinds of URLs.

if I'm guessing right we could replace that check with one that verifies that the requesting user can _read_ the blog even if they aren't a member of it.

**Comment by lancewillett** [comment]
Good find! Yes, that is to limiting... as a WordPress.com user I'd like to follow lots of sites where I'm not a member (user on the site).

In fact, in the social scope of reading WordPress.com sites — the "Following" action is more likely on sites I don't belong to already.

Agreed on the better check: if the requesting user can _read_ the blog they can likely _follow_ it also.

**Comment by dmsnell** [comment]
After some more investigation it doesn't look like that code _should_ be blocking this. This may take more exploration on why it's occurring.

**Comment by lancewillett** [comment]
OK. Thinking about this more, the "is member of blog?" would only apply to non-public blogs. Such as a private site, which is "followable" by its members only.

**Comment by dmsnell** [comment]
> the "is member of blog?" would only apply to non-public blogs. Such as a private site, which is "followable" by its members only.

Right. The problem I hit is that when I updated the code in my sandbox to fix it things didn't work the way I expected. There are multiple caches in play which _could_ explain some of this.

**Comment by lancewillett** [comment]
Closing; the method to follow via Likes seems to have updated several times since I last tested this. The links say "Subscribe" now—and it's hard to tell if this is fixed.

If I see this again I'll consider re-opening.
